src/pages/Lending/hooks/useIsSweepNeededQuery.ts

```diff
@@ -0,0 +1,57 @@
+import type { AssetId } from '@shapeshiftoss/caip'
+import { fromAssetId } from '@shapeshiftoss/caip'
+import { useQuery } from '@tanstack/react-query'
+import { useCallback, useMemo } from 'react'
+import { getChainAdapterManager } from 'context/PluginProvider/chainAdapterSingleton'
+import { bnOrZero } from 'lib/bignumber/bignumber'
+import { isUtxoChainId } from 'state/slices/portfolioSlice/utils'
+
+type UseIsSweepNeededQueryProps = {
+  assetId: AssetId
+  address: string | null
+  amountCryptoBaseUnit: string
+  txFeeCryptoBaseUnit: string
+  enabled: boolean
+}
+
+export const useIsSweepNeededQuery = ({
+  assetId,
+  address,
+  amountCryptoBaseUnit,
+  txFeeCryptoBaseUnit,
+  enabled,
+}: UseIsSweepNeededQueryProps) => {
+  const useIsSweepNeededQueryKey = useMemo(
+    () => ['isSweepNeeded', { assetId, address, amountCryptoBaseUnit, txFeeCryptoBaseUnit }],
+    [assetId, address, amountCryptoBaseUnit, txFeeCryptoBaseUnit],
+  )
+
+  const getIsSweepNeeded = useCallback(
+    async (assetId: AssetId, address: string) => {
+      const adapter = getChainAdapterManager().get(fromAssetId(assetId).chainId)
+      if (!adapter) return
+      const { chainId } = fromAssetId(assetId)
+      // Chains others than UTXO don't require a sweep step
+      if (!isUtxoChainId(chainId)) return false
+
+      const addressAccount = await adapter?.getAccount(address)
```

Comment:
We wouldn't have notion of potential unconfirmed balances for this address if there was something in the mempool, though we probably shouldn't make too many decisions based on mempool state however. Think this is the safest check